### PR TITLE
Spack package for cice4

### LINKED ISSUE #87: Spack package for cice4
Create a Spack package for `cice4` based on  [ACCESS-NRI/cice4](https://github.com/ACCESS-NRI/cice4) and [cice5](https://github.com/ACCESS-NRI/spack-packages/blob/main/packages/cice5/package.py).

The `Makefile` in https://github.com/penguian/access-esm-build-gadi uses the  [ACCESS-NRI/cice4](https://github.com/ACCESS-NRI/cice4) repository on GitHub to build CICE4.1 for ACCESS-ESM1.5.

### PR DESCRIPTION
**Pull request by penguian**
This is a minimal viable product merge for the `access-esm1.5` version of `cice4`.
Closes #87

### CODE REVIEW

**Comment by penguian** [comment]
Both files include references to their sources. The command
```
spack install cice4%intel@19.0.5.281
```
successfully build an executable that runs the ACCESS-ESM 1.5 `pre-industrial` configuration successfully, with bitwise identical output to previous runs, excluding timing.

File: packages/cice4/package.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+CPPFLAGS   := -P -traditional
+CPPDEFS    := -DLINUX -DPAROPT
+CFLAGS     := {CFLAGS}
+FIXEDFLAGS := -132
+FREEFLAGS  :=
+"""
+
+        config["gcc"] = """
+FFLAGS = -Wall -fdefault-real-8 -fdefault-double-8 -ffpe-trap=invalid,zero,overflow -fallow-argument-mismatch
+LDFLAGS    := $(FFLAGS)
```
Comment (harshula) [inline packages/cice4/package.py:85]: Add a comment explaining why.

File: packages/cice4/package.py [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+# https://spack.readthedocs.io/en/latest/build_systems/makefilepackage.html
+class Cice4(MakefilePackage):
+    """The Los Alamos sea ice model (CICE) is the result of an effort to develop a computationally efficient sea ice component for a fully coupled atmosphere-land global climate model."""
+
+    homepage = "https://www.access-nri.org.au"
+    git = "https://github.com/ACCESS-NRI/cice4.git"
+
+    maintainers("penguian")
+
+    version("master", branch="ESM_1.5")
```
[the remark this answers is not in the document]
Comment (penguian) [reply packages/cice4/package.py]: Both version and branch are now "access-esm1.5"

File: packages/cice4/package.py (hunk 1 above)
Comment (penguian) [reply packages/cice4/package.py:87]: Done.

**Comment by penguian** [comment]
Rebase to `main` created a merge conflict in `packages/cice4/package.py`. I resolved this.
I also re-tested with `pre-industrial` configuration and output is still bit identical.

**Comment by harshula** [comment]
I've removed the unnecessary rebase merge commit.

**Review by harshula — APPROVED** [review]
Please squash merge the PR and make the commit message as concise as possible.

**Pull request merged**